File: nodejs/01-dirs.js
/*
 * This tutorial shows basic directory operations in Zbox.
 */

const Zbox = require('@zbox/nodejs');

(async function run() {
  // create a Zbox instance
  const zbox = new Zbox();

  // initialise Zbox environment and turn on debug logs
  await zbox.initEnv({ debug: true });

  // open the repo
  var repo = await zbox.openRepo({
    uri: '[your_repo_uri]',
    pwd: 'secret password',
    opts: { create: true }
  });

  // create a dir
  await repo.createDir('/dir');

  // recursively create a nested directory
  await repo.createDirAll('/foo/bar/baz');

  // list root dir
  const root = await repo.readDir('/');
  console.log(root);

  // list /foo dir
  const foo = await repo.readDir('/foo');
  console.log(foo);

  // remove an empty dir
  await repo.removeDir('/dir');

  // remove a dir and all its children
  await repo.removeDirAll('/foo');

  // close repo and exit Zbox
  await repo.close();
  await zbox.exit();
})();

File: rust/src/bin/01-dirs.rs
/*
 * This tutorial shows basic directory operations in Zbox.
 */

extern crate zbox;

use zbox::{init_env, RepoOpener};

fn main() {
    // initialise zbox environment, called first
    init_env();

    // create and open a repository
    let mut repo = RepoOpener::new()
        .create(true)
        .open("[your_repo_uri]", "secret password")
        .unwrap();

    // create a dir
    repo.create_dir("/dir").unwrap();

    // recursively create a nested directory
    repo.create_dir_all("/foo/bar/baz").unwrap();

    // list root dir
    let root = repo.read_dir("/").unwrap();
    println!("{:?}", root);

    // list /foo dir
    let foo = repo.read_dir("/foo").unwrap();
    println!("{:?}", foo);

    // remove an empty dir
    repo.remove_dir("/dir").unwrap();

    // remove a dir and all its children
    repo.remove_dir_all("/foo").unwrap();
}
